language_service/src/completion/tests.rs

```diff
@@ -821,6 +821,44 @@ fn notebook_block() {
     );
 }
 
+#[test]
+fn notebook_auto_open_start_of_cell() {
+    check_notebook(
+        &[
+            ("cell1", "namespace Foo { operation Bar() : Unit {} }"),
+            ("cell2", "↘"),
+        ],
+        &["Fake"],
+        &expect![[r#"
+            [
+                Some(
+                    CompletionItem {
+                        label: "Fake",
+                        kind: Function,
+                        sort_text: Some(
+                            "0800Fake",
+                        ),
+                        detail: Some(
+                            "operation Fake() : Unit",
+                        ),
+                        additional_text_edits: Some(
+                            [
+                                (
+                                    Span {
+                                        start: 0,
+                                        end: 0,
+                                    },
```

Comment:
Good point, I don't know why I just went with the most trivial test case here... I manually tested the more complicated stuff, but I'll add the test cases to be safe.

Since we're looking at the source contents and not the ast, the first non-whitespace character (ignoring `%%qsharp` which is erased before we get here) counts as the beginning of the cell.

If you have a cell _with_ a namespace, you'll still get the `open` statement at the top of the cell, above the namespace. Frankly it did not seem worth special-casing. I did attempt to implement it that way, but it ended up being more complicated than you'd think.